module/zcommon/zfs_fletcher_aarch64_neon_impl.S

```diff
@@ -0,0 +1,113 @@
+/*
+ * Optimize the neon implementation which is generated by GCC12
+ * or Clang by the commit: https://github.com/openzfs/zfs/pull/14234
+
+ * Due to the compiler limitation, the GCC version before 12 will
+ * produce not optimized code for Aarch64, so use the asm function
+ * to accelerate the performance here.
+ * Copyright (C) 2022 Kevin Zhao.
+ *
+ * Authors:
+ *	Kevin Zhao <kevin.zhao@linaro.org>
```

Comment:
This has been so heavily revised based on my feedback that I am wondering if you would be willing to list me as a co-author on this file to reflect the development history. It would be fine to put my name second.